src/simulation/elements/TSNS.cpp

```diff
@@ -82,7 +84,35 @@ int Element_TSNS::update(UPDATE_FUNC_ARGS)
 					continue;
 				if ((r&0xFF)!=PT_TSNS && (r&0xFF)!=PT_METL && parts[r>>8].temp > parts[i].temp)
 					parts[i].life = 1;
+				if ((r & 0xFF) != PT_FILT && ((r&0xFF) != PT_TSNS && parts[i].tmp == 1))
```

Comment:
This should be

if (parts[i].tmp == 1 && (r&0xFF) != PT_TSNS && (r&0xFF) != PT_FILT)